### PR TITLE
Remove unnecessary parsing logic.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
I believe this was a hold-over from the old native compiler and when we originally did generics.  

Specifically, this change makes it so that when we're parsing out Type-*Arguments* we do not have special code looking for Type-*Parameter*-Constraints.  Constraints are not relevant in this context.  They do not appear with, or near, type-arguments (which is just a separated list of Types, between ```<>```).  

Note: there is similar code in our Type-*Parameter* parsing logic.  There it makes sense as someone might have ```class C<T,    where T : I``` and we don't want to consume ```where``` as a type-*parameter*.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler @jcouv @cston

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler  @gafter to take a look.

Thanks!

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -5454,11 +5454,6 @@ private PostSkipAction SkipBadTypeArgumentListTokens(SeparatedSyntaxListBuilder<
         // Parses the individual generic parameter/arguments in a name.
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: Perhaps adjust the comment (remove "parameter/")

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler Can i get a second set of eyes.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz Improvements in parser that benefit IDE and compiler.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: Sure.

**Comment by MattGertz** [comment]
Approved pending tests.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Pull request merged**